commons/src/main/java/org/infinispan/commons/marshall/WrappedByteArray.java

```diff
@@ -74,7 +73,7 @@ public boolean equalsWrappedBytes(WrappedBytes other) {
 
    @Override
    public int hashCode() {
-      return hashCode;
+      return hashCode == 0 ? (hashCode = Arrays.hashCode(bytes)) : hashCode;
```

Comment:
Maybe we shouldn't cache the hash code at all? `MurmurHash3.hash(Object)` doesn't use it anyway...